tests/e2e/vsan_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This test is going to create volume on the fresh testbed very first time.
+// After installing vmdk volume plugin/driver, volume creation should not be
+// failed very first time.
+
+// This test is going to cover various vsan related test cases
+
+package e2e
+
+import (
+	"os"
+	"strings"
+
+	"github.com/vmware/docker-volume-vsphere/tests/utils/admincli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/govc"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+
+	. "gopkg.in/check.v1"
+)
+
+type VsanTestSuite struct {
+	hostIP     string
+	esxIP      string
+	vsanDSName string
+	volumeList []string
+}
+
+func (s *VsanTestSuite) SetUpSuite(c *C) {
+	s.hostIP = os.Getenv("VM2")
+	s.esxIP = os.Getenv("ESX")
+
+	dsNameList := govc.GetDatastoreList()
+	s.vsanDSName = ""
+	for _, ds := range dsNameList {
+		if strings.HasPrefix(ds, "vsan") {
```

Comment:
this can work ... more robust and correct approach is to look for datastore type using govc.